nemo/collections/asr/losses/rnnt.py

```diff
@@ -281,19 +281,21 @@ def forward(self, log_probs, targets, input_lengths, target_lengths):
             targets = targets.narrow(dim=1, start=0, length=max_targets_len)
 
         # Loss reduction can be dynamic, so set it prior to call
-        if self.reduction != 'mean_batch':
+        if self.reduction in ['mean', 'sum']:  # underlying losses can do mean or sum
             self._loss.reduction = self.reduction
 
         # Compute RNNT loss
         loss = self._loss(acts=log_probs, labels=targets, act_lens=input_lengths, label_lens=target_lengths)
 
         # Loss reduction can be dynamic, so reset it after call
-        if self.reduction != 'mean_batch':
+        if self.reduction in ['mean', 'sum']:
             self._loss.reduction = 'none'
 
-        # Loss reduction only for mean_batch mode
+        # Loss reduction only for mean_batch and mean_volume modes
         if self.reduction == 'mean_batch':
             loss = torch.mean(loss)
+        elif self.reduction == 'mean_volume':
+            loss = torch.sum(loss) / torch.sum(target_lengths)
```

Comment:
Should the underlying loss reduction be disabled for this case (seems to be mean by default)? Also, would this work properly with the fused batches?

Reply:
To the best of my understanding, the RNNT Joint [does disable ](https://github.com/NVIDIA/NeMo/blob/61ddd23eba8d88cc08d2f2f7909b75f5333cd4d1/nemo/collections/asr/modules/rnnt.py#L1330)the underlying loss reduction so that it can do its own calculation